packages/shared/src/components/PostOptionsMenu.tsx

```diff
@@ -495,11 +500,51 @@ export default function PostOptionsMenu({
   postOptions.push({
     icon: <MenuIcon Icon={BlockIcon} />,
     label: isSourceBlocked
-      ? `Show posts from ${post?.source?.name}`
-      : `Don't show posts from ${post?.source?.name}`,
+      ? `Unblock ${post?.source?.name}`
+      : `Block ${post?.source?.name}`,
     action: isSourceBlocked ? onUnblockSourceClick : onBlockSourceClick,
   });
 
+  if (post?.author && post?.author?.id !== user?.id) {
+    postOptions.push({
+      icon: <MenuIcon Icon={BlockIcon} />,
+      label: isBlockedAuthor
+        ? `Unblock ${post.author.name}`
+        : `Block ${post.author.name}`,
+      action: async () => {
+        const clearCache = () => {
```

Comment:
Maybe best to abstract this above in it's own function for readability?
(non blocking)